akka-javasdk/src/main/scala/akka/javasdk/impl/agent/AgentImpl.scala

```diff
@@ -159,6 +169,24 @@ private[impl] final class AgentImpl[A <: Agent](
 
   }
 
+  private def toSpiContextMessages(conversationHistory: ConversationHistory): Vector[SpiAgent.ContextMessage] = {
+    import scala.jdk.CollectionConverters._
+
+    conversationHistory
+      .messages()
+      .asScala
+      .flatMap {
+        case m if m.isInstanceOf[AiMessage] =>
+          Some(new SpiAgent.ContextMessage.AiMessage(m.asInstanceOf[AiMessage].text()))
+        case m if m.isInstanceOf[UserMessage] =>
+          Some(new SpiAgent.ContextMessage.UserMessage(m.asInstanceOf[UserMessage].text()))
+        case m =>
+          log.warn("Unsupported message type [{}], ignoring", m.getClass.getName)
+          None
```

Comment:
why is this graceful ignore needed? aren't we in control of all types, and they are sealed?
if it is some other type we can throw IllegalStateException, since it's a bug?